pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -771,28 +809,27 @@ func isPodRunnableOrSucceeded(p *corev1.Pod) bool {
 // lastActiveTime returns the last timestamp on which the pod was observed active:
 // - the time the pod was declared Failed
 // - the deletion time
-func lastActiveTime(p *corev1.Pod) time.Time {
-	lastTransition := metav1.Now()
+func lastActiveTime(clock clock.Clock, p *corev1.Pod) time.Time {
```

Comment:
The logic of this function seems to be changed. For example :
```
deletionTimestamp=now-1min
lastTransition=now-2min
```
Then, we would return lastTransition, meaning `now-2min`. Now we are returning `now-1min`. I'm not sure which one is better, but I prefer to de-couple changing the logic from cleanup. 

If you think the logic warrants to be improved then issue a separate PR for it.